ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/insert-values.json

```diff
@@ -15,7 +15,7 @@
         {"topic": "test_topic", "timestamp": 1234, "key": "key", "value": {"ID": 10}}
       ],
       "responses": [
-        {"admin": {"@type": "currentStatus", "statementText": "{STATEMENT}"}}
```

Comment:
Unless the feature flag is enabled, this test fails if this check is present since the statement in the response contains `KEY_FORMAT` whereas the original statement text does not. Once we enable the feature flag by default, and the test is updated to include `KEY_FORMAT` in the request, this can be added back in. Same for the other changes in this file.

If desirable, we can add a version of these tests with the feature flag enabled and the statementText response check present, but I'm not sure how much value we gain from these particular checks so that doesn't seem worth it.

Reply:
Another issue caused by use regenerating the SQL :(

The more I think about this, the less I think we need the injector.  The query plan is generated from a combination of the statement and the config, so we have all the information we need to build the plan _without_ injecting the key format. Do that, and all these concerns go away. 

I've knocked together a quick example of what I mean in a draft PR, so take a look: https://github.com/confluentinc/ksql/pull/6254

Reply:
As discussed in https://github.com/confluentinc/ksql/pull/6254 and offline, we're going to keep the injector since it makes sense for the engine to require key and value formats are present (on CS/CT statements), similar to how it requires schemas be present. As a convenience, however, ksqlDB has a mechanism that allows users to not need to explicitly supply schemas and key/value formats in some cases (schema inference, and via default key/value format configs). These conveniences are handled before the statement enters the engine, which is the role of the injectors. By populating default formats once upfront, the various places in the engine (and in later injectors) that require key and value formats to be present do not each have to handle filling in defaults from the config.